python/ray/_private/utils.py

```diff
@@ -1258,7 +1258,13 @@ def init_grpc_channel(
             private_key=private_key,
             root_certificates=ca_cert,
         )
-        channel = grpc_module.secure_channel(address, credentials, options=options)
+        
```

Comment:
I have one high level question:

Is this needed only when you want to run the CLIs remotely instead of within the cluster. Like what if you can ssh to the head node and run there, do you still need the custom metadata?

Reply:
Yes that is correct, but only admins can ssh onto the Head Node. We have a reverse proxy in front of Ray which acts as an authentication layer meaning it would have to be passed as a header. There were quite a few companies at Ray Summit that have a similar architecture, so think this would be a helpful feature.

Reply:
Yea, I agree we need to implement it.

Reply:
Hope that clarifies. Let me know if you have any other questions!

Reply:
@architkulkarni Do all cli command use the same gRPC stub? I believe we could wrap the gRPC stub additional auth if the environment variable exists. If so, can you point out the specific stub call that gets called? Is it the one posted above? Appreciate all the discussion, just want to make sure I am understanding everything correctly.

Reply:
Some CLI commands use http like `ray job sumit` instead of grpc.

Reply:
Right. But for the ones that do use gRPC, do they use the same stubs?

Reply:
I think the stubs are `internal_kv_*()` (e.g. `internal_kv_get()`, `internal_kv_list()`). I actually take a look of all the CLIs. I think the ones that we want to run remotely and need authentications are mostly http based CLIs. So I think we should start with those first.